### DISCUSSION TITLE
response_model_exclude in nested objects

### DISCUSSION THREAD

**Discussion by Erhannis** [fastapi/fastapi#2824] [Questions] [answered]
Is it possible to use `response_model_exclude` (and possibly `response_model_include`) for nested structures?  Suppose I have a file approximately like
```
class Thing(BaseModel):
    name: str
    otherthing: OtherThing

class OtherThing(BaseModel):
    name: str
    color: str
    hugeblob: str

@app.get("/test", response_model=Thing, response_model_exclude={"otherthing.hugeblob"}) # example possible syntax
async def read_test_item():
    return test_item # test_item from somewhere else in the app
```
Now, this doesn't work as phrased.  (I tried.)  Nor does `...exclude={"hugeblob"}`.  Is there some other (fairly easy) way to omit particular fields on a child?

**Comment by tiangolo** [comment, marked as answer]
`response_model_exclude` is not recommended, it's there only because some people kept asking for it, but defining a model that has more data than what will actually be sent will be misleading and end up incorrectly documented in OpenAPI.

The right approach would be to declare a model that doesn't include what you don't want, and inherit from that to add the fields that you need to include in other places.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
